src/agent/coverage/src/disasm.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::BTreeSet;
+
+use anyhow::Result;
+use iced_x86::{Decoder, DecoderOptions, Instruction};
+
+use crate::code::{ModuleIndex, Symbol};
+
+pub struct ModuleDisassembler<'a> {
+    module: &'a ModuleIndex,
+    data: &'a [u8],
+}
+
+impl<'a> ModuleDisassembler<'a> {
+    pub fn new(module: &'a ModuleIndex, data: &'a [u8]) -> Result<Self> {
+        Ok(Self { module, data })
+    }
+
+    /// Find block entry points for every symbol in the module.
+    pub fn find_blocks(&self) -> BTreeSet<u64> {
+        let mut blocks = BTreeSet::new();
+
+        for symbol in self.module.symbols.iter() {
+            let sym_blocks = self.find_symbol_blocks(symbol);
+            blocks.extend(sym_blocks);
```

Comment:
Maybe `find_symbol_blocks` should modify an existing BTreeSet instead of constructing a new BTreeSet.

Reply:
I like that. We could always wrap a mutating method in one that returns a `BTreeSet`, if we wanted.

Reply:
As exposed, this function could actually panic, if an invalid `Symbol` was passed. Fixing that, too.